packages/ui/src/accounts/hooks/useGroupLocks/helpers.ts

```diff
@@ -18,19 +18,16 @@ export const useIsWGLockRecoverable = (hasWGLock: boolean, stakingAccount: strin
   })
   const workerId = workerData?.workers[0]?.id
   const { unstakingPeriodEnd } = useWorkerUnstakingPeriodEnd(workerId)
+  const status = applications[0]?.status
 
-  if (applications[0]?.status === 'ApplicationStatusPending') {
-    return true
-  }
-
-  if (applications[0]?.status === 'ApplicationStatusAccepted') {
+  if (status === 'ApplicationStatusAccepted') {
     if (unstakingPeriodEnd) {
       return Date.parse(unstakingPeriodEnd) > Date.now()
     }
     return true
```

Comment:
@WRadoslaw could you remind me why if there is no unstaking period the stake is recoverable and why is the unstaking period happening when the application is accepted please ? I don't remember how this is supposed to work...

Reply:
Good question. I think this logic was inherit from Rafal's lock refactor PR, but now when I look at it, it seems invalid. Since `useWorkerUnstakingPeriodEnd` will return a falsy value if the worker didn't leave/got kicked.

Unstaking period happens only when the given application was already accepted, so the applicant is now a worker.   More [here](https://joystream.gitbook.io/testnet-workspace/system/working-groups#worker).

```suggestion
      return Date.parse(unstakingPeriodEnd) > Date.now()
    }
    return false
```